joint_state_publisher_gui/joint_state_publisher_gui/joint_state_publisher_gui.py

```diff
@@ -209,20 +219,53 @@ def randomize_event(self, event):
         self.randomize()
 
     def randomize(self):
-        rospy.loginfo("Randomizing")
+        self.jsp.get_logger().info("Randomizing")
         for name, joint_info in self.joint_map.items():
             joint = joint_info['joint']
             joint_info['slider'].setValue(
-                    self.valueToSlider(random.uniform(joint['min'], joint['max']), joint))
-
-    def sliderUpdate(self, event):
-        for name, joint_info in self.joint_map.items():
-            joint_info['slidervalue'] = joint_info['slider'].value()
-        self.update_sliders()
+                self.valueToSlider(random.uniform(joint['min'], joint['max']), joint))
 
     def valueToSlider(self, value, joint):
         return (value - joint['min']) * float(RANGE) / (joint['max'] - joint['min'])
 
     def sliderToValue(self, slider, joint):
         pctvalue = slider / float(RANGE)
         return joint['min'] + (joint['max']-joint['min']) * pctvalue
+
+    def closeEvent(self, event):
+        self.running = False
+
+    def loop(self):
+        while self.running:
+            rclpy.spin_once(self.jsp, timeout_sec=0.1)
+
+
+def main(input_args=None):
+    if input_args is None:
+        input_args = sys.argv
+
+    # Initialize rclpy with the command-line arguments
+    rclpy.init(args=input_args)
+
+    # Strip off the ROS 2-specific command-line arguments
+    stripped_args = rclpy.utilities.remove_ros_args(args=input_args)
+    parser = argparse.ArgumentParser()
+    parser.add_argument('urdf_file', help='URDF file to use', nargs='?', default=None)
+
+    # Parse the remaining arguments, noting that the passed-in args must *not*
+    # contain the name of the program.
+    parsed_args = parser.parse_args(args=stripped_args[1:])
+
+    app = QApplication(sys.argv)
+    jsp_gui = JointStatePublisherGui('Joint State Publisher',
+                                     JointStatePublisher(parsed_args.urdf_file))
```

Comment:
I launched the GUI with no urdf via CLI, and then published it via `/robot_description`, but the sliders don't show up. Is something supposed to update the GUI when that happens?

Reply:
Yep, this was another missing feature.  Thanks for catching.  Fixed in 2832411.